controllers/hcloudmachinetemplate_controller.go

```diff
@@ -65,20 +67,17 @@ func (r *HCloudMachineTemplateReconciler) Reconcile(ctx context.Context, req rec
 	log = log.WithValues("HCloudMachineTemplate", klog.KObj(machineTemplate))
 
 	// Fetch the Cluster.
-	cluster, err := util.GetOwnerCluster(ctx, r.Client, machineTemplate.ObjectMeta)
-	if cluster == nil {
-		_, err := utils.GetOwnerClusterClass(ctx, r.Client, machineTemplate.ObjectMeta)
-		if err != nil {
-			return reconcile.Result{Requeue: true}, fmt.Errorf("failed to retrieve ownerReference as clusterClass: %w", err)
+	var cluster *clusterv1.Cluster
+	_, err := getOwnerClusterClass(ctx, r.Client, machineTemplate.ObjectMeta)
```

Comment:
i like it better if you actually do it like this:
if a cluster class is found, return. 
else (you don't need it in fact because you return before): continue